java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspTemplateUI.java

```diff
@@ -273,9 +286,48 @@ private static String singleSelection(List<QuickPickItem> selectedGroups) {
         }
         return selectedGroups.get(0).getUserData().toString();
     }
+    
+    private static final String ATTR_SOURCE_GROUP = "new.sourceGroup";  // NOI18N
+    private static final String ATTR_SOURCE_HINT = "new.sourceHint"; // NOI18N
```

Comment:
We should start documenting these [magic strings](http://wiki.apidesign.org/wiki/MagicalStrings) somewhere. Especially when we start spreading them outside of the `java.lsp.server` module.

Even if we don't, it should be `<arch stability="private"..../>` as that is an API someone can depend on (but shouldn't).

Reply:
Arch document created and magic strings described in d56b619f25.